File: backend/dataset/rotate/rotate_15.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n=len(nums)
        k=k%n
        nums.reverse()
        nums[0:k]=reversed(nums[0:k])
        nums[k:]=reversed(nums[k:])
        

File: backend/CDataset/rotate/rotate_15.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        vector<int> v;
        int n = nums.size();
        
    
        if(k==0 || n == 1 ) {
            return;
        }
        
        if(k >= n) {
            k = k-n;
        }
        
        for(int j = n-k; j < n; j++) {
               v.push_back(nums[j]);
           }
        
         for(int z = 0; z < n-k; z++) {
               v.push_back(nums[z]);
           }
        
      
        for(int i = 0; i < n; i++) {
            nums[i] = v[i];
        }
    }
};
